### PR TITLE
vrp: allow supervisord to open its log file

### PR DESCRIPTION
**Pull request by akonradi**
Commit Message: Allow supervisord to open its log file
Additional Description:
Change the default location of the log file and give supervisord
permissions to write to it.

Risk Level: low
Testing: built image locally
Docs Changes: n/a
Release Notes: n/a
Platform Specific Features: n/a

### CODE REVIEW

**Comment by mattklein123** [comment]
cc @phlax who has been working on similar things.

File: configs/google-vrp/supervisor.conf [hunk 1]
```diff
@@ -1,5 +1,6 @@
 [supervisord]
 nodaemon=true
+logfile=/var/log/supervisor/supervisord.log
```
Comment (htuch) [inline configs/google-vrp/supervisor.conf:3]: I'm thinking this could just be `/dev/stdout`, similar to the application logs below.

**Review by htuch — COMMENTED** [review]
Thanks @akonradi for the quick fix! Did you notice this makes a difference at HEAD? The reason I ask is that I replayed https://www.envoyproxy.io/docs/envoy/latest/intro/arch_overview/security/google_vrp#working-with-the-docker-images instructions and it seemed to work on `-latest` but I saw the supervisord errors at fixed release versions.

File: configs/google-vrp/supervisor.conf (hunk 1 above)
Comment (akonradi) [reply configs/google-vrp/supervisor.conf:3]: Looks like that doesn't work. I get the following when trying to run with output to /dev/stdout:
```
2020-11-17 22:53:39,487 CRIT could not write pidfile /supervisord.pid
Traceback (most recent call last):
  File "/usr/bin/supervisord", line 11, in <module>
    load_entry_point('supervisor==3.3.1', 'console_scripts', 'supervisord')()
  File "/usr/lib/python2.7/dist-packages/supervisor/supervisord.py", line 365, in main
    go(options)
  File "/usr/lib/python2.7/dist-packages/supervisor/supervisord.py", line 375, in go
    d.main()
  File "/usr/lib/python2.7/dist-packages/supervisor/supervisord.py", line 84, in main
    self.run()
  File "/usr/lib/python2.7/dist-packages/supervisor/supervisord.py", line 100, in run
    self.options.write_pidfile()
  File "/usr/lib/python2.7/dist-packages/supervisor/options.py", line 1123, in write_pidfile
    self.logger.critical('could not write pidfile %s' % self.pidfile)
  File "/usr/lib/python2.7/dist-packages/supervisor/loggers.py", line 287, in critical
    self.log(LevelsByName.CRIT, msg, **kw)
  File "/usr/lib/python2.7/dist-packages/supervisor/loggers.py", line 293, in log
    handler.emit(record)
  File "/usr/lib/python2.7/dist-packages/supervisor/loggers.py", line 186, in emit
    self.doRollover()
  File "/usr/lib/python2.7/dist-packages/supervisor/loggers.py", line 211, in doRollover
    if not (self.stream.tell() >= self.maxBytes):
IOError: [Errno 29] Illegal seek
```

**Comment by akonradi** [comment]
Yeah I only tried this for the current master. How would I apply it to a release version? Just use that as the base docker image?

File: ci/Dockerfile-envoy-google-vrp [hunk 2]
```diff
@@ -16,6 +16,7 @@ ADD configs/google-vrp/supervisor.conf /etc/supervisor.conf
 ADD test/config/integration/certs/serverkey.pem /etc/envoy/certs/serverkey.pem
 ADD test/config/integration/certs/servercert.pem /etc/envoy/certs/servercert.pem
 # ADD %local envoy bin% /usr/local/bin/envoy
+RUN chmod 777 /var/log/supervisor
```
Comment (htuch) [inline ci/Dockerfile-envoy-google-vrp:19]: I think this also needs:
```
+RUN chmod a+r /etc/supervisor.conf /etc/envoy/* /etc/envoy/certs/*
+RUN chmod a+rx /usr/local/bin/launch_envoy.sh
```

**Review by htuch — COMMENTED** [review]
Thanks. I've confirmed this fixes things locally. I think if you can deal with the permission issues as well, we can ship this and it should fix `latest`. I'll do some followups to deal with released version images and putting the VRP image runtime validation into CI to avoid snafus like this in future.

**Review by htuch — APPROVED** [review]
LGTM, thanks!

**Comment by htuch** [comment]
Coverage failure is unrelated, force merging.

**Comment by cpakulski** [comment]
Backported to releases 1.16 and 1.15. Removing `backport/approved` label.

**Pull request merged**